### PR TITLE
Feature 2347 pulse survey UI

### PR DESCRIPTION
**Pull request by mvolkmann**

### CODE REVIEW

File: web-ui/package.json [hunk 1]
```diff
@@ -103,7 +103,7 @@
     "typescript-eslint": "^7.7.1",
     "vite": "^5.2.6",
     "vite-tsconfig-paths": "^4.3.2",
-    "vitest": "^1.4.0",
+    "vitest": "^1.6.0",
```
Comment (mvolkmann) [inline web-ui/package.json:106]: Upgraded to the latest version.

File: web-ui/src/components/pulse/Pulse.jsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
+import PropTypes from 'prop-types';
```
Comment (mvolkmann) [inline web-ui/src/components/pulse/Pulse.jsx:1]: This is a new component that displays a title, a row of sentiment icon buttons, and a text input for entering a comment.

File: web-ui/src/components/pulse/Pulse.css [hunk 3]
```diff
@@ -0,0 +1,31 @@
+.pulse {
```
Comment (mvolkmann) [inline web-ui/src/components/pulse/Pulse.css:1]: CSS for the new `Pulse` component.

File: web-ui/src/components/pulse/Pulse.test.jsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
+import React from 'react';
```
Comment (mvolkmann) [inline web-ui/src/components/pulse/Pulse.test.jsx:1]: Tests for new `Pulse` component.

File: web-ui/src/pages/PulsePage.css [hunk 5]
```diff
@@ -0,0 +1,14 @@
+.pulse-page {
```
Comment (mvolkmann) [inline web-ui/src/pages/PulsePage.css:1]: CSS for new `PulsePage` component.

File: web-ui/src/pages/PulsePage.jsx [hunk 6]
```diff
@@ -1,11 +1,147 @@
-import React from 'react';
```
Comment (mvolkmann) [inline web-ui/src/pages/PulsePage.jsx:1]: Updated the placeholder `PulsePage` component with the intended content described in this story.

File: web-ui/src/pages/PulsePage.jsx [hunk 7]
```diff
@@ -1,11 +1,147 @@
...
+    } catch (err) {
+      console.error('PulsePage.jsx submit:', err);
+    }
+  };
+
+  return (
+    <div className="pulse-page">
+      {submittedToday ? (
+        <Typography className="submitted-today" variant="h6">
+          Thank you for submitting your pulse today!
```
Comment (mvolkmann) [inline web-ui/src/pages/PulsePage.jsx:114]: Michael may be providing new copy text soon.

File: web-ui/src/pages/PulsePage.test.jsx [hunk 8]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
```
Comment (mvolkmann) [inline web-ui/src/pages/PulsePage.test.jsx:1]: Snapshot test for new `PulsePage` component.

File: web-ui/vite.config.ts [hunk 9]
```diff
@@ -1,23 +1,23 @@
-import { defineConfig } from "vitest/config"
```
Comment (mvolkmann) [inline web-ui/vite.config.ts:1]: Prettier reformatted this.

File: web-ui/vite.config.ts [hunk 10]
```diff
@@ -1,23 +1,23 @@
...
   plugins: [react()],
   test: {
     globals: true,
-    environment: "jsdom",
-    setupFiles: "./src/setupTests.js",
-    css: true,
-    reporters: ["verbose"],
+    environment: 'jsdom',
+    setupFiles: './src/setupTests.js',
+    css: false,
```
Comment (mvolkmann) [inline web-ui/vite.config.ts:12]: Setting `css` to `false` is the only real change made in this file. Before this change we were getting the error "Could not parse CSS stylesheet" for any CSS files imported by a component that use nested CSS rules. It seems we don't really need to parse the CSS in order for all our existing tests to pass. I verified that.

**Review by jackkeller — APPROVED** [review]
Approved 🎉 💥

**Review by vhscom — APPROVED** [review]
LGTM. I noticed we default users currently to dissatisfied. In a later iteration perhaps we don't default the user's settings, to help prevent them from submitting before they have explicitly selected internal/external pulse ratings.

**Comment by mvolkmann** [comment]
From Michael ...
Screenshot 2024-05-16 at 5 00 07 PM

**Pull request merged**